src/lib/optimizer/join_ordering/join_edge.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <iostream>
+#include <ostream>
+
+#include "boost/dynamic_bitset.hpp"
+
+namespace opossum {
+
+class AbstractJoinPlanPredicate;
+
+/**
+ * Represents a (hyper)edge in a JoinGraph. Each predicate must operate exactly on the vertices in vertex_set
+ */
+class JoinEdge final {
+ public:
+  explicit JoinEdge(const boost::dynamic_bitset<>& vertex_set);
```

Comment:
Created a `using JoinVertexSet = boost::dynamic_bitset<>` and added the comment

"We use a dynamic bitset here, since it has good support for operations commonly performed on join vertex sets, such as UNION, INTERSECTION, etc."